Crash on Device Rotation

### Expected

The app does not crash when you rotate the device either portrait to landscape or landscape to portrait.
### Observed

The app crashes when the device is rotated either portrait to landscape or landscape to portrait immediately after launch.  The steps to reproduce the issue and stack trace are below.
#### Steps
1. Open the app.
2. Rotate the device.
3. Notice the crash.

```
10-28 13:54:28.957 519-519/com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.amaze.filemanager, PID: 519
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.amaze.filemanager.utils.color.ColorPreference.getColorAsString(com.amaze.filemanager.utils.color.ColorUsage)' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2649)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2710)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4486)
        at android.app.ActivityThread.-wrap19(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1467)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6088)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
        Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.amaze.filemanager.utils.color.ColorPreference.getColorAsString(com.amaze.filemanager.utils.color.ColorUsage)' on a null object reference
        at com.amaze.filemanager.fragments.Main.onCreate(Main.java:205)
        at android.support.v4.app.Fragment.performCreate(Fragment.java:1951)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1029)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1252)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1234)
        at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:2041)
        at android.support.v4.app.FragmentController.dispatchCreate(FragmentController.java:163)
        at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:332)
        at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:84)
        at com.amaze.filemanager.activities.BasicActivity.onCreate(BasicActivity.java:30)
        at com.amaze.filemanager.activities.BaseActivity.onCreate(BaseActivity.java:38)
        at com.amaze.filemanager.activities.MainActivity.onCreate(MainActivity.java:241)
        at android.app.Activity.performCreate(Activity.java:6679)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2602)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2710) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4486) 
        at android.app.ActivityThread.-wrap19(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1467) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6088) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
```
#### Tested

Google Pixel on Android 7.1 with Amaze File Manager 3.1.2 Beta 2

--- a/src/main/java/com/amaze/filemanager/activities/BasicActivity.java
+++ b/src/main/java/com/amaze/filemanager/activities/BasicActivity.java
@@ -1,8 +1,6 @@
 package com.amaze.filemanager.activities;
 
-import android.os.Bundle;
 import android.preference.PreferenceManager;
-import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
 
 import com.amaze.filemanager.utils.Futils;
@@ -13,23 +11,28 @@ import com.amaze.filemanager.utils.color.ColorPreference;
  * Created by rpiotaix on 17/10/16.
  */
 public class BasicActivity extends AppCompatActivity implements UtilitiesProviderInterface {
+    private boolean initialized = false;
     protected ColorPreference colorPreference;
     private Futils utils;
 
+    private void initialize() {
+        utils = new Futils();
+
+        colorPreference = ColorPreference.loadFromPreferences(this, PreferenceManager.getDefaultSharedPreferences(this));
+    }
+
     @Override
     public Futils getFutils() {
+        if (!initialized)
+            initialize();
+
         return utils;
     }
 
     public ColorPreference getColorPreference() {
+        if (!initialized)
+            initialize();
+
         return colorPreference;
-    }
-
-    @Override
-    protected void onCreate(@Nullable Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        utils = new Futils();
-
-        colorPreference = ColorPreference.loadFromPreferences(this, PreferenceManager.getDefaultSharedPreferences(this));
     }
 }

